core/src/test/java/org/infinispan/api/AsyncAPITest.java

```diff
@@ -91,6 +97,26 @@ public void testPutAsync() throws Exception {
       assertEquals("v2", c.get("k"));
    }
 
+   public void testPutAsyncEntry() throws Exception {
+      Metadata metadata = new EmbeddedMetadata.Builder().version(new NumericVersion(1)).build();
+      CompletableFuture<CacheEntry<String, String>> f = c.getAdvancedCache().putAsyncEntry("k", "v1", metadata);
+      assertFutureResult(f, null);
+      assertEquals("v1", c.get("k"));
+
+      f = c.getAdvancedCache().putAsyncEntry("k", "v2", EmbeddedMetadata.EMPTY);
+      assertFutureResultOn(f, previousEntry -> {
+         assertEquals("k", previousEntry.getKey());
+         assertEquals("v1", previousEntry.getValue());
+         assertNotNull(previousEntry.getMetadata());
+
+         Metadata previousMetadata = previousEntry.getMetadata();
+         assertEquals(InequalVersionComparisonResult.EQUAL, previousMetadata.version().compareTo(new NumericVersion(1)));
```

Comment:
Should be able to use just a simple `assertEquals(new NumericVersion(1), previousMetadata.version());` as they implement equality.